### PR TITLE
Added typing to contrastive, focal_loss, giou_loss

### PR DESCRIPTION
**Pull request by abhichou4**
Added typing to contrastive loss, focal loss and giou loss (#989)

### CODE REVIEW

**Comment by abhichou4** [comment]
@googlebot I signed it!

> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> memo **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> #### What to do if you already signed the CLA
> ##### Individual signers
> 
>     * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> 
> ##### Corporate signers
> 
>     * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> 
>     * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
>     * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> 
> information_source **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Ftensorflow%2Faddons%2Fpull%2F1007) for more info**.

@googlebot I signed it!

File: tensorflow_addons/losses/contrastive.py [hunk 1]
```diff
@@ -88,17 +92,19 @@ class ContrastiveLoss(tf.keras.losses.Loss):
...
     def __init__(
         self,
-        margin=1.0,
-        reduction=tf.keras.losses.Reduction.SUM_OVER_BATCH_SIZE,
-        name="contrasitve_loss",
-    ):
+        margin: Number = 1.0,
+        reduction: tf.keras.losses.Reduction = tf.keras.losses.Reduction.SUM_OVER_BATCH_SIZE,
+        name:str = "contrasitve_loss",
+    ) -> None:
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/losses/contrastive.py]: When using `__init__` you don't have to specify the return type because it's obvious.

**Comment by gabrieldemarmiesse** [comment]
Please take a look at the pre-commit hook to fix the black and pep8 errors: https://github.com/tensorflow/addons/blob/master/STYLE_GUIDE.md#python

File: tensorflow_addons/losses/giou_loss.py [hunk 2]
```diff
@@ -48,26 +51,30 @@ class GIoULoss(tf.keras.losses.Loss):
...
+                 mode: str = 'giou',
+                 reduction: tf.keras.losses.Reduction = tf.keras.losses.Reduction.AUTO,
+                 name: str = 'giou_loss') -> None:
         if mode not in ['giou', 'iou']:
             raise ValueError("Value of mode should be 'iou' or 'giou'")
         super().__init__(name=name, reduction=reduction)
         self.mode = mode
 
-    def get_config(self):
+    def get_config(self) -> None:
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/losses/giou_loss.py]: It returns a dict it seems.

File: tensorflow_addons/losses/contrastive.py [hunk 3]
```diff
@@ -88,17 +92,19 @@ class ContrastiveLoss(tf.keras.losses.Loss):
 
     def __init__(
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/losses/contrastive.py]: When adding types to `__init__` functions, you should add the `@typechecked` decorator. You can look at other classes in the codebase for reference.

**Comment by gabrieldemarmiesse** [comment]
Very nice! Thanks!
